tests/basis_test.py

```diff
@@ -67,6 +67,58 @@ def test_cheb_call_random_points():
         f = ChebyshevFirstKind(n)
         assert numpy.isclose(f(x),special.eval_chebyt(n,x))
 
+
+def test_cheb_derivative():
+    """Test if the correct derivative is generated."""
+    f0 = ChebyshevFirstKind(0)
+    f1 = ChebyshevFirstKind(1)
+    f2 = ChebyshevFirstKind(2)
+    x0, x1 = 1, -0.5
+    assert numpy.isclose(f0.derivative(x0), 0)
+    assert numpy.isclose(f1.derivative(x0), 1)
+    assert numpy.isclose(f2.derivative(x0), 4)
+    assert numpy.isclose(f0.derivative(x1), 0)
+    assert numpy.isclose(f1.derivative(x1), 1)
+    assert numpy.isclose(f2.derivative(x1), -2)
+
+
+def test_cheb_derivative_polynomial():
+    """Test if the correct object is created for derivation."""
+    f2 = ChebyshevFirstKind(2)
+    derivative_polynomial = f2._derivative_polynomial
+    f2_derivative = f2.derivative(0.5)
+    assert derivative_polynomial is None
+    assert isinstance(f2._derivative_polynomial, ChebyshevSecondKind)
+    assert numpy.isclose(f2._derivative_polynomial(0.5), 1)
+
+
+def test_cheb_2nd_derivative():
+    """Test if the correct derivative is generated."""
+    u0 = ChebyshevSecondKind(0)
+    u1 = ChebyshevSecondKind(1)
+    u2 = ChebyshevSecondKind(2)
+    x0, x1, x2 = 1, -1, 0.5
+    assert numpy.isclose(u0.derivative(x0), 0)
+    assert numpy.isclose(u1.derivative(x0), 2)
+    assert numpy.isclose(u2.derivative(x0), 8)
+    assert numpy.isclose(u0.derivative(x1), 0)
+    assert numpy.isclose(u1.derivative(x1), 2)
+    assert numpy.isclose(u2.derivative(x1), -8)
+    assert numpy.isclose(u0.derivative(x2), 0)
+    assert numpy.isclose(u1.derivative(x2), 2)
+    assert numpy.isclose(u2.derivative(x2), 4)
```

Comment:
Long lists like this are hard to read, why not do this instead?
```suggestion
    assert numpy.isclose(u0.derivative(1), 0)
    assert numpy.isclose(u1.derivative(1), 2)
    assert numpy.isclose(u2.derivative(1), 8)

    assert numpy.isclose(u0.derivative(-1), 0)
    assert numpy.isclose(u1.derivative(-1), 2)
    assert numpy.isclose(u2.derivative(-1), -8)

    assert numpy.isclose(u0.derivative(0.5), 0)
    assert numpy.isclose(u1.derivative(0.5), 2)
    assert numpy.isclose(u2.derivative(0.5), 4)
```
Same comment for the other tests